jbpm-task-assigning/jbpm-task-assigning-core/src/main/java/org/jbpm/task/assigning/model/Task.java

```diff
@@ -0,0 +1,339 @@
+        super(id);
+        this.processInstanceId = processInstanceId;
+        this.processId = processId;
+        this.containerId = containerId;
+        this.name = name;
+        this.priority = priority;
+        this.inputData = inputData;
+    }
+
+    public long getProcessInstanceId() {
+        return processInstanceId;
+    }
+
+    public void setProcessInstanceId(long processInstanceId) {
+        this.processInstanceId = processInstanceId;
+    }
+
+    public String getProcessId() {
+        return processId;
+    }
+
+    public void setProcessId(String processId) {
+        this.processId = processId;
+    }
+
+    public String getContainerId() {
+        return containerId;
+    }
+
+    public void setContainerId(String containerId) {
+        this.containerId = containerId;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public int getPriority() {
+        return priority;
+    }
+
+    public void setPriority(int priority) {
+        this.priority = priority;
+    }
+
+    public Map<String, Object> getInputData() {
+        return inputData;
+    }
+
+    public void setInputData(Map<String, Object> inputData) {
+        this.inputData = inputData;
+    }
+
+    public boolean isPinned() {
+        return pinned;
+    }
+
+    public void setPinned(boolean pinned) {
+        this.pinned = pinned;
+    }
+
+    public boolean isPublished() {
+        return published;
+    }
+
+    public void setPublished(boolean published) {
+        this.published = published;
+    }
+
+    /**
+     * @return the set of OrganizationalEntities that are enabled for executing this task.
+     */
+    public Set<OrganizationalEntity> getPotentialOwners() {
+        return potentialOwners;
+    }
+
+    public void setPotentialOwners(Set<OrganizationalEntity> potentialOwners) {
+        this.potentialOwners = potentialOwners;
+    }
+
+    /**
+     * @return the set of labels for this task.
+     */
+    public Set<TypedLabel> getTypedLabels() {
+        return typedLabels;
+    }
+
+    public void setTypedLabels(Set<TypedLabel> typedLabels) {
+        this.typedLabels = typedLabels;
+    }
+
+    public TaskOrUser getPreviousTaskOrUser() {
+        return previousTaskOrUser;
+    }
+
+    public void setPreviousTaskOrUser(TaskOrUser previousTaskOrUser) {
+        this.previousTaskOrUser = previousTaskOrUser;
+    }
+
+    @Override
+    public User getUser() {
+        return user;
+    }
+
+    public void setUser(User user) {
+        this.user = user;
+    }
+
+    public Integer getStartTime() {
+        return startTime;
+    }
+
+    public void setStartTime(Integer startTime) {
+        this.startTime = startTime;
+    }
+
+    @Override
+    public Integer getEndTime() {
+        return endTime;
+    }
+
+    public void setEndTime(Integer endTime) {
+        this.endTime = endTime;
+    }
+
+    public int getDuration() {
```

Comment:
Could we use `Duration` here? We will then avoid the problem of what unit this is in.